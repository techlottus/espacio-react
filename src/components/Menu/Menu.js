import React, { useEffect } from "react";



export const Menu = React.memo(({ data, onClose, onCard,outside }) => {
  const menuRef = React.createRef();

  useEffect(() => {
    menuRef.current.data = {
      items: data.items || [],
      disabledBtnClose: data.disabledBtnClose,
      isShow: data.isShow,
    };
  },[data]) // eslint-disable-line react-hooks/exhaustive-deps


  useEffect(() => {
    menuRef.current.addEventListener("onClose", onClose);
    menuRef.current.addEventListener("onCard", onCard);
    const handleClickOutside = (event) => {
      if (menuRef.current && !menuRef.current.contains(event.target) && window.innerWidth > 991) {
        outside();
      }
      
    }
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-menu ref={menuRef}></lottus-menu>
    </>
  );
});
