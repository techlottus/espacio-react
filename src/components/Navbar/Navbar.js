import React, { useEffect } from "react";



export const Navbar = React.memo(({ data, onRight, onLogo }) => {

  const navbarRef = React.createRef();


  useEffect(() => {
    navbarRef.current.data = {
      urlLogo: data.urlLogo || "",
      iconLeft: data.iconLeft || "",
      iconsRight: data.iconsRight || [],
      zindex: 10
    };
  }, [data]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    navbarRef.current.addEventListener("onRight", onRight);
    navbarRef.current.addEventListener("onLogo", onLogo);

  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-navbar ref={navbarRef}></lottus-navbar>
    </>
  );
});
