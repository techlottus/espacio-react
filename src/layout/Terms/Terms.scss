@import "../../App.scss";

.navbar {
  position: relative;
  .menu {
    width: 100%;
    position: fixed;
    right: 0;
    top: 0;
    z-index: 1000;
    height: 100vh;
    background-color: map-get($palette-overlay, b-50);

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      height: 100%;
      background-color: transparent;
      right: pixeltorem(16px);
      top: pixeltorem(80px);
      width: auto;
      
    }

    @media only screen and (min-width: map-get($grid-breakpoints, xl)) {
      height: 100%;
      background-color: transparent;
      right: pixeltorem(16px);
      width: auto;
      top: pixeltorem(116px);
    }
  }
}

.terms-view{
  @extend .headerapp-top;
}
.desktop-container {
  display: flex;

  .termscontent{ 
    width: 33.333%;
    left:0;
    height: 100vh;
  }
    .termsprivacywrap {
      width: 66.666%;
      background-color: map-get($palette-neutral, n05);
    }
  }


.termsfeedback {
  @extend .alertfeedbackcmp;
}