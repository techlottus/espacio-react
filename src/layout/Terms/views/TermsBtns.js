import React, { useEffect, useState } from "react";
import { Button } from "../../../components/Button/Button";
import { ProgressBar } from "../../../components/ProgressBar/ProgressBar";
import { Toggle } from "../../../components/Toggle/Toggle";
import {
  dataTerms,
  dataPrivacy,
  dataProgress,
  sendBtn,
  toggleData,
  typesTermsAndPrivacy,
} from "../../../constants/Terms.constant";

import { useDispatch, useSelector } from "react-redux";
import { useHistory } from "react-router";

import "./TermsBtns.scss";
import { patchPrivacyService } from "../../../middlewares/termsMiddelware";
import { sendInfoTM } from "../../../tagging/services/sendInfoTagManager";
import { switchOnTermsTag } from "../../../tagging/flows/termsTag";
import { testTermsTitle } from "../../../constants/test/testTerms";
const TermsBtns = ({ onClickTerms, onClickPrivacy }) => {

  const { texts } = useSelector((state) => state.texts);

  const [btnTerms, btnInitTerms] = useState({
    ...dataTerms,
    icon: "",
    disabled: true,
    title: texts?.terms.btns.btnTerms,
  });
  const [btnPrivacy, btnInitPrivacy] = useState({
    ...dataPrivacy,
    icon: "",
    title: texts?.terms.btns.btnPrivacy,
  });
  const [btnSend, setBtnSend] = useState({
    ...sendBtn,
    disabled: true,
    title: texts?.terms.btns.btnEnter,
  });
  const [isToogle, setIsToogle] = useState({
    ...toggleData
  });


  const [progressData, setProgressData] = useState({
    ...dataProgress,
    progress: 0,
    description: texts?.terms.progress.progressZero,
  });

  const [nextPrivacy, setNextPrivacy] = useState(false);
  const [nextTerms, setNextTerms] = useState(false);

  const termsStore = useSelector((state) => state.terms);

  const history = useHistory();

  const dispatch = useDispatch();

  useEffect(() => {
    window.scrollTo(0, 0);
    if (termsStore.privacy_notice_agreed) {
      btnInitPrivacy({ ...dataPrivacy, icon: "done", disabled: true });
      setProgressData({
        ...dataProgress,
        progress: 50,
        description: texts?.terms.progress.progressMiddle,
      });
      btnInitTerms({ ...dataTerms, icon: "", disabled: false });
    }
    if (termsStore.terms_and_conditions_agreed) {
      btnInitTerms({ ...dataTerms, icon: "done", disabled: true });
      setProgressData({
        ...dataProgress,
        progress: 100,
        description: texts?.terms.progress.progressComplete,
      });
    }
    if (
      termsStore.privacy_notice_agreed &&
      termsStore.terms_and_conditions_agreed
    ) {
      setBtnSend({
        ...btnSend,
        disabled: false,
      });
    }
  }, [termsStore]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    if (nextPrivacy) {
      onClickPrivacy();
    }
  }, [nextPrivacy]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    if (nextTerms) {
      onClickTerms();
    }
  }, [nextTerms]); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <div className="terms-container">
      <h2 className="termsbtns-title">{texts?.terms.title}</h2>
      <p className="termsbtns-text">{texts?.terms.subTitle}</p>

      <div className="buttons">
        <div className="button-wrap">
          <Button
            key="privacy-btn"
            data={{
              ...btnPrivacy,
              test: testTermsTitle.btnPrivacyAction
            }}
            onClick={() => setNextPrivacy((state) => !state)}
          />
        </div>
        <div className="button-wrap second-btn">
          <Button
            data={{
              ...btnTerms,
              test: testTermsTitle.btnTermsAction
            }}
            onClick={() => setNextTerms((state) => !state)}
          />
        </div>
      </div>
      <div className="newsletter-wrap">
        <Toggle
          data={{
            ...isToogle,
            test: testTermsTitle.toogleTerms
          }}
          onSwitch={(e) => {
            setIsToogle({...isToogle, status:e.detail});
            e.detail ? sendInfoTM(window, switchOnTermsTag, 'on') :  sendInfoTM(window, switchOnTermsTag, 'off')
          }}
        />
        <p className="termsbtns-text">{texts?.terms.toggle}</p>
      </div>
      <div className="progress-bar">
        <ProgressBar data={progressData} />
      </div>
      <div className="send-btn">
        <Button
          data={{
            ...btnSend,
            test: testTermsTitle.btnFinish
          }}
          onClick={() => {
            dispatch(
              patchPrivacyService(
                typesTermsAndPrivacy.promotions,
                history,
                null
              )
            );
          }}
        />
      </div>
    </div>
  );
};

export default TermsBtns;
