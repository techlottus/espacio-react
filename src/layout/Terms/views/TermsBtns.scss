@import '../../../App.scss';

.terms-container {
  padding: pixeltorem(48px) pixeltorem(5px);
  margin: 0 pixeltorem(15px);
  font-family: "Nunito", sans-serif;

  @media only screen and (min-width: map-get($grid-breakpoints, m)) {
    padding: pixeltorem(48px) pixeltorem(5px) 0 pixeltorem(5px);

  }

  .termsbtns-title{
    font-family: 'Bebas',sans-serif;
    font-size: pixeltorem(42px);
    padding-bottom: pixeltorem(5px);

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
          font-size: pixeltorem(56px);
      }
    
  }
  .termsbtns-text {
    font-size: pixeltorem(14px);
    line-height: normal;
  }

  .buttons {
    padding: pixeltorem(15px) 0 pixeltorem(10px) 0;

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      padding: pixeltorem(20px) 0;
    }

    .button-wrap {
        padding: pixeltorem(8px) 0;

        button{
          width: 100%;
          text-align: center;
          justify-content: center;

          @media only screen and (min-width: map-get($grid-breakpoints, m)) {
            width: auto;
           }
        }
    }
  }
  .newsletter-wrap{
    display: flex;
    align-items: center;

    > .termsbtns-text{
      padding-left: pixeltorem(14px);
    }
  }

  .progress-bar{
    padding: pixeltorem(36px) 0 pixeltorem(15px) 0;

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      padding: pixeltorem(35px) 0;
    }
  }

  .send-btn{
      button{
        width: 100%;
        text-align: center;
        justify-content: center;

        @media only screen and (min-width: map-get($grid-breakpoints, m)) {
         width: auto;
        }
      }
  }
}
