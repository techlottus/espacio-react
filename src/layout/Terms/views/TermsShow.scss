@import "../../../App.scss";

.terms-page{

  .terms-page-container {
    padding: pixeltorem(90px) pixeltorem(5px) 0  pixeltorem(5px);
    margin: 0 pixeltorem(15px);
    font-family: "Nunito", sans-serif; 

    @extend .fixedcontainer-height;

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      margin: 0 pixeltorem(35px) 0 pixeltorem(60px);
      font-family: "Nunito", sans-serif;
      padding-top: pixeltorem(45px);
      height: 100%;
    }

    .terms-arrow {
      display: block;
      padding-bottom: pixeltorem(20px);
      color: palette-colors($brand, p00);

      @media only screen and (min-width: map-get($grid-breakpoints, m)) {
        display: none; 
      }
    }
  
    .terms-content {
      max-height: pixeltorem(320px);
      padding-right: pixeltorem(40px);
      overflow-y: scroll;
      > .termshow-title {
        font-family: 'Bebas',sans-serif;
        font-size: pixeltorem(24px);
        padding-bottom: pixeltorem(16px);

        @media only screen and (min-width: map-get($grid-breakpoints, m)) {
          font-size: pixeltorem(56px);
        }
      }
      .terms-map {
        > .termshow-subtitle {
          padding: pixeltorem(5px) 0;
          font-size: pixeltorem(16px);
          font-weight: 700;
        }
  
        .termshow-texts{
           > .termshow-text {
            font-size: pixeltorem(12px);
            padding: pixeltorem(10px) 0;
          }
    
          > .termshow-ul {
            .termshow-li {
              padding: pixeltorem(5px) 0;
              font-size: pixeltorem(12px);
    
              &::before {
                content: "\2022";
                color: map-get($palette-neutral, n80);
                font-weight: bold;
                display: inline-block;
                width: 1em;
                margin-left: 1em;
              }
            }
          }
          > .termshow {
            padding: pixeltorem(5px) 0;
            font-size: pixeltorem(12px);  
          }
          .termshow-ol-number {
            .termshow-li-number {
              padding: pixeltorem(5px) 0;
              font-size: pixeltorem(12px);
              list-style-type: auto;
              margin-left: 4em;
            }
          }
           .termshow-ol-upperroman {
            .termshow-li-upperroman {
              padding: pixeltorem(5px) 0;
              font-size: pixeltorem(12px);
              list-style-type: upper-roman;
              margin-left: 5em;
            }
          }
          .termshow-ul-circle {
            .termshow-li-circle {
              padding: pixeltorem(5px) 0;
              font-size: pixeltorem(12px);
              list-style-type: circle;
              margin-left: 5em;
            }
          }
          .termshow-ul-square {
            .termshow-li-square {
              padding: pixeltorem(5px) 0;
              font-size: pixeltorem(12px);
              list-style-type: square;
              margin-left: 5em;
            }
          }
          .termshow-ul-guion {
            .termshow-li-guion {
              padding: pixeltorem(5px) 0;
              font-size: pixeltorem(12px);
              margin-left: 3em;

              &::before {
                content: "-";
                margin-right: 5px;
              }
            }
          }
        }
      }
    }
  
    .terms-feed {
      padding-top: pixeltorem(20px);

      @media only screen and (min-width: map-get($grid-breakpoints, m)) {
        padding-top: pixeltorem(60px);
      }
    }
  
    .terms-btns {
      padding: pixeltorem(20px) 0;

      @media only screen and (min-width: map-get($grid-breakpoints, m)) {
        display: flex;
        justify-content: flex-end;
      }

      .button-wrap {
        padding: pixeltorem(8px) 0;

        @media only screen and (min-width: map-get($grid-breakpoints, m)) {
          padding: 0 pixeltorem(10px);
        }

        button{
          width: 100%;
          text-align: center;
          justify-content: center;
        }
      }
    }
  }
}