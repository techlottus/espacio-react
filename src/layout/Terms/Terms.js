import React, { useState, useEffect } from "react";
import { TermsContext } from "./TermsContext";
import TermsBtns from "./views/TermsBtns";
import TermsShow from "./views/TermsShow";
import { Navbar } from "../../components/Navbar/Navbar";
import { Menu } from "../../components/Menu/Menu";
import Welcome from "./views/Welcome";
import {
  privacyContent,
  termsContent,
  dataNavBar,
  dataMenu,
  typesTermsAndPrivacy,
  typeTermsBtn,
} from "../../constants/Terms.constant";
import "./Terms.scss";
import { Switch, useHistory, Route, Redirect } from "react-router-dom";
import {
  getPrivacyService,
  patchPrivacyService,
} from "../../middlewares/termsMiddelware";
import { useDispatch, useSelector } from "react-redux";
import { resetTerms, setErrorTerms } from "../../actions/termsAction";
import { typesRequestErrors } from "../../constants/error.constant";
import { setTokens } from "../../actions/loginAction";
import { setAuth } from "../../actions/headersAction";
import env from "../../enviroment/environment";
import { notiObs } from "../../observables/notificationObs";
import { typesNoti } from "../../types/typeNoti";
import {
  headerLogoHeaderTag,
  headerMenuOpenHeaderTag,
} from "../../tagging/flows/headerTag";
import { sendInfoTM } from "../../tagging/services/sendInfoTagManager";
import { privacyBtnTermsTag } from "../../tagging/flows/termsTag";
import { getImageOfAssetsMark } from "../../helpers/getImages";

const Terms = () => {
  const history = useHistory();

  const [menu] = useState({
    ...dataMenu,
    disabledBtnClose: window.innerWidth < 991,
  });
 
  const termsStore = useSelector((state) => state.terms);
  const { texts } = useSelector((state) => state.texts);

  const dispatch = useDispatch();

  const [isModal, setIsModal] = useState(false);

  const onRight = (e) => {
    switch (e.detail) {
      case "apps":
        setIsModal((prevIsModal) => !prevIsModal);
        break;
      case "notifications":
        break;
      default:
        break;
    }
  };

  const [desktop, setDesktop] = useState(window.innerWidth > 991);
  const [terms, showTerms] = useState({
    terms: false,
    privacy: false,
    init: true,
  });

  const clickTerms = () => {
    if (!termsStore.terms_and_conditions_agreed) {
      showTerms({
        terms: true,
        privacy: false,
      });
    }
  };

  const clickPrivacy = () => {
    if (!termsStore.privacy_notice_agreed) {
      showTerms({
        terms: false,
        privacy: true,
      });
    }
  };

  const redirectTerms = () => {
    if (window.innerWidth < 991 && !termsStore.terms_and_conditions_agreed) {
      history.push("/terms/terms-and-conditions");
    }
  };

  const redirectPrivacy = () => {
    if (window.innerWidth < 991 && !termsStore.privacy_notice_agreed) {
      history.push("/terms/advice-privacy");
      sendInfoTM(window, privacyBtnTermsTag);
    }
  };

  const clickShow = (typeBtn, type) => {
    switch (typeBtn) {
      case typeTermsBtn.accept:
        dispatch(patchPrivacyService(type, history, showTerms));
        break;
      case typeTermsBtn.decline:
        showTerms({
          terms: false,
          privacy: false,
          init: true,
        });
        history.replace("/terms");
        break;
      default:
        break;
    }
  };

  const updateMedia = () => {
    setDesktop(window.innerWidth > 991);
  };

  useEffect(() => {
    window.addEventListener("resize", updateMedia);
    return () => window.removeEventListener("resize", updateMedia);
  });

  useEffect(() => {
    dispatch(getPrivacyService(history));

    return () => {
      dispatch(resetTerms());
      dispatch(
        setErrorTerms(typesRequestErrors.patchPrivacyNotice, false, null)
      );
      dispatch(setErrorTerms(typesRequestErrors.getPrivacyNotice, false, null));
    };
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    if (
      termsStore.errors &&
      termsStore.errors[typesRequestErrors.patchPrivacyNotice].isError
    ) {
      notiObs.next({
        type: typesNoti.error,
        text: termsStore.errors[typesRequestErrors.patchPrivacyNotice].msg,
      });
    }

    if (
      termsStore.errors &&
      termsStore.errors[typesRequestErrors.getPrivacyNotice].isError
    ) {
      notiObs.next({
        type: typesNoti.error,
        text: termsStore.errors[typesRequestErrors.getPrivacyNotice].msg,
      });
    }
  }, [termsStore.errors]);

  const onLogo = () => {
    history.push("/terms");
  };
  const welcomeMsg = () => {
    if(terms.init ){
          if (
      termsStore.privacy_notice_agreed &&
      termsStore.terms_and_conditions_agreed
    ) {
      return (
        <Welcome
          key="init"
          title={texts?.terms.welcomeTitle.welcomeThanks}
          image={getImageOfAssetsMark(texts?.terms?.images?.thankyouTerms)}
        />
      );
    } else {
      return (
        <Welcome
          key="end"
          title={texts?.terms.welcomeTitle.welcomeIntro}
          image={getImageOfAssetsMark(texts?.terms?.images?.welcomeTerms)}
        />
      );
    }
    }
  };

  const privacyView = () => {
    if(terms.privacy){
      return (
        <TermsShow
          title={privacyContent.title}
          contents={privacyContent.contents}
          type={typesTermsAndPrivacy.privacy}
          clickShow={clickShow}
        />
      );
    }

  };

  const termsView = () => {
    if(terms.terms){
      return (
        <TermsShow
          title={termsContent.title}
          contents={termsContent.contents}
          type={typesTermsAndPrivacy.terms}
          clickShow={clickShow}
        />
      );
    }
  };

  return (
    <>
      <TermsContext.Provider value={null}>
        <div className="navbar">
          <Navbar
            data={dataNavBar}
            onRight={(e) => {
              onRight(e);
              sendInfoTM(window, headerMenuOpenHeaderTag, "open");
            }}
            onLogo={() => {
              onLogo();
              sendInfoTM(window, headerLogoHeaderTag);
            }}
          />
          {isModal ? (
            <div className="menu">
              <Menu
                data={menu}
                onCard={(e) => {
                  if (e.detail === "back") {
                    dispatch(setTokens(null, null));
                    dispatch(setAuth(null));
                    sessionStorage.removeItem(env.campusvirtual_token);
                    window.location.href = env.logout;
                  }
                }}
                onClose={() => {
                  setIsModal(!isModal);
                  sendInfoTM(window, headerMenuOpenHeaderTag, "close");
                }}
                outside={() => {
                  setIsModal(false);
                  sendInfoTM(window, headerMenuOpenHeaderTag, "close");
                }}
              />
            </div>
          ) : (
            ""
          )}
        </div>
        <Switch>
          <Route path="/terms/advice-privacy">
            <TermsShow
              title={privacyContent.title}
              contents={privacyContent.contents}
              type={typesTermsAndPrivacy.privacy}
              clickShow={clickShow}
            />
          </Route>
          <Route path="/terms/terms-and-conditions">
            <TermsShow
              title={termsContent.title}
              contents={termsContent.contents}
              type={typesTermsAndPrivacy.terms}
              clickShow={clickShow}
            />
          </Route>
          <Route exact path="/terms/">
            <div className="terms-view">
              {desktop ? (
                <div className="desktop-container">
                  <div className="termscontent">
                    <TermsBtns
                      onClickTerms={clickTerms}
                      onClickPrivacy={() => {
                        clickPrivacy();
                        sendInfoTM(window, privacyBtnTermsTag);
                      }}
                    />
                  </div>
                  <div className="termsprivacywrap">
                    { welcomeMsg()}
                    {privacyView()}
                    {termsView()}
                  </div>
                </div>
              ) : (
                <TermsBtns
                  onClickTerms={redirectTerms}
                  onClickPrivacy={redirectPrivacy}
                />
              )}
            </div>
          </Route>
          <Redirect to="/terms/" />
        </Switch>
      </TermsContext.Provider>
    </>
  );
};

export default Terms;
