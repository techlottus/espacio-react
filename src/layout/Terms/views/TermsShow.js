import React, { useState } from "react";
import Notification from "../views/Notification";
import { Button } from "../../../components/Button/Button";
import { termsBtns } from "../../../constants/Terms.constant";

import "./TermsShow.scss";
import { Link } from "react-router-dom";
import { texts } from "../../../texts/indexText";
import { sendInfoTM } from "../../../tagging/services/sendInfoTagManager";
import {
  privacyBtnDeclinedTermsTag,
  termsBtnDeclinedTermsTag,
} from "../../../tagging/flows/termsTag";
import { testTermsTitle } from "../../../constants/test/testTerms";

const TermsShow = ({ title, contents, clickShow, type }) => {
  const [btns] = useState([...termsBtns]);

  return (
    <div className="terms-page">
      <div className="terms-page-container">
        <Link to="/terms">
          <div className="terms-arrow">
            <span className="material-icons">arrow_back</span>
          </div>
        </Link>
        <div className="terms-content">
          <h2 className="termshow-title">{title}</h2>
          {contents.map((content, i) => {
            return (
              <div className="terms-map" key={i}>
                <h3 className="termshow-subtitle">{content.subtitle}</h3>
                <div
                  className="termshow-texts"
                  dangerouslySetInnerHTML={{ __html: content.text }}
                ></div>
              </div>
            );
          })}
        </div>
        <div className="terms-feed">
          <Notification icon={"info"} text={texts?.terms.notification} />
        </div>
        <div className="terms-btns">
          {btns.map((data, index) => {
            return (
              <div className="button-wrap" key={index}>
                <Button
                  data={{
                    ...data,
                    test: testTermsTitle.btnTermsAndPrivacyAction + '-' +type + '-' +data.id
                  }}
                  onClick={() => {
                    clickShow(data.id, type);
                    type === "privacy" && data.id === "decline"
                      ? sendInfoTM(
                          window,
                          privacyBtnDeclinedTermsTag,
                          "noacepto"
                        )
                      : sendInfoTM(
                          window,
                          privacyBtnDeclinedTermsTag,
                          "acepto"
                        );
                    type === "terms" && data.id === "decline"
                      ? sendInfoTM(window, termsBtnDeclinedTermsTag, "noacepto")
                      : sendInfoTM(window, termsBtnDeclinedTermsTag, "acepto");
                  }}
                />
              </div>
            );
          })}
        </div>
      </div>
    </div>
  );
};

export default TermsShow;
