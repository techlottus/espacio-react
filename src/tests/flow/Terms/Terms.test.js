import React from "react";
import { mount } from "enzyme";
import Terms from "../../../layout/Terms/Terms";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { typesRequestErrors } from "../../../constants/error.constant";
import { texts } from "../../../texts/indexText";
import { BrowserRouter } from "react-router-dom/cjs/react-router-dom.min";
import { act } from "@testing-library/react";
import TermsBtns from "../../../layout/Terms/views/TermsBtns";
import { termsText } from "../../../texts/termsText";
import { Navbar } from "../../../components/Navbar/Navbar";
import { Menu } from "../../../components/Menu/Menu";
import TermsShow from "../../../layout/Terms/views/TermsShow";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

window.scrollTo = jest.fn();
window.dataLayer = {
  push: jest.fn(),
};
window.innerWidth = 1200;

let initState = {
  terms: {
    privacy_notice_agreed: false,
    terms_and_conditions_agreed: false,
    errors: {
      [typesRequestErrors.getPrivacyNotice]: {
        isError: false,
        msg: null,
      },
      [typesRequestErrors.patchPrivacyNotice]: {
        isError: false,
        msg: null,
      },
    },
  },
  texts: {
    terms: {
      ...termsText,
    },
  },
};

let store = mockStore(initState);
const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };

const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <BrowserRouter>
        <Terms history={historyMock} />
      </BrowserRouter>
    </Provider>
  );
};

describe("Test <Terms/>", () => {
  beforeEach(() => {
    store = mockStore(initState);
  });

  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });

  describe("should show errors", () => {
    test("path privacy notcie", () => {
      initState = {
        ...initState,
        terms: {
          ...initState.terms,
          errors: {
            ...initState.terms.errors,
            [typesRequestErrors.patchPrivacyNotice]: {
              isError: true,
              msg: "Error",
            },
          },
        },
      };
      const wrapper = handleMount(initState);
    });
    test("get privacy notice", () => {
      initState = {
        ...initState,
        terms: {
          ...initState.terms,
          errors: {
            ...initState.terms.errors,
            [typesRequestErrors.getPrivacyNotice]: {
              isError: true,
              msg: "Error",
            },
          },
        },
      };
      const wrapper = handleMount(initState);
    });
  });

  describe("should ", () => {
    test("1", () => {
      initState = {
        ...initState,
        terms: {
          ...initState.terms,
          privacy_notice_agreed: false,
        },
      };
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find(TermsBtns).props().onClickPrivacy();
      });
      wrapper.update();
      act(() => {
        wrapper.find(TermsShow).first().props().clickShow("accept");
      });
      act(() => {
        wrapper.find(TermsShow).first().props().clickShow("decline");
      });
      act(() => {
        wrapper.find(TermsShow).first().props().clickShow("other");
      });
      wrapper.update();
    });
    test("2", () => {
      initState = {
        ...initState,
        terms: {
          ...initState.terms,
          terms_and_conditions_agreed: false,
        },
      };
      const wrapper = handleMount(initState);

      act(() => {
        wrapper.find(TermsBtns).props().onClickTerms();
      });
    });
  });

  describe("should click on navbar", () => {
    test("should show menu", () => {
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find(Navbar).props().onRight({
          detail: "apps",
        });
      });
      wrapper.update();
      act(() => {
        wrapper.find(".menu").find(Menu).props().onCard({
          detail: "back",
        });
      });
      wrapper.update();
      act(() => {
        wrapper.find(".menu").find(Menu).props().onCard({
          detail: "other",
        });
      });
      wrapper.update();
      act(() => {
        wrapper.find(".menu").find(Menu).props().outside();
      });
      act(() => {
        wrapper.find(".menu").find(Menu).props().onClose();
      });
      wrapper.update();
      act(() => {
        wrapper.find(Navbar).props().onRight({
          detail: "notifications",
        });
      });
      wrapper.update();
      act(() => {
        wrapper.find(Navbar).props().onRight({
          detail: "other",
        });
      });
      wrapper.update();
    });
    test("should click on logo", () => {
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find(Navbar).props().onLogo();
      });
      wrapper.update();
    });
  });

  test("Should redirect to terms", () => {
    window.innerWidth = 990;

    const wrapper = handleMount(initState);

    act(() => {
      wrapper.find(TermsBtns).props().onClickTerms();
    });

    act(() => {
      wrapper.find(TermsBtns).props().onClickPrivacy();
    })

  });

});
