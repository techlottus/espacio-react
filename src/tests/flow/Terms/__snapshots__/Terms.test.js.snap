// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Terms/> should init component 1`] = `
Array [
  <div
    className="navbar"
  >
    <lottus-navbar />
  </div>,
  <div
    className="terms-view"
  >
    <div
      className="desktop-container"
    >
      <div
        className="termscontent"
      >
        <div
          className="terms-container"
        >
          <h2
            className="termsbtns-title"
          />
          <p
            className="termsbtns-text"
          />
          <div
            className="buttons"
          >
            <div
              className="button-wrap"
            >
              <lottus-button />
            </div>
            <div
              className="button-wrap second-btn"
            >
              <lottus-button />
            </div>
          </div>
          <div
            className="newsletter-wrap"
          >
            <lottus-toogle-switch />
            <p
              className="termsbtns-text"
            />
          </div>
          <div
            className="progress-bar"
          >
            <lottus-progress-bar />
          </div>
          <div
            className="send-btn"
          >
            <lottus-button />
          </div>
        </div>
      </div>
      <div
        className="termsprivacywrap"
      >
        <div
          className="welcome-container"
        >
          <div
            className="title-container"
          >
            <h2
              className="welcome-title"
            />
          </div>
          <div
            className="img-container"
          >
            <img
              alt="logo-notebook"
              className="img-notebook"
              src="images/Terms/terms.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
